runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -1220,7 +1220,9 @@ else if (type.equals(EXPIRY_SIGNAL_NAME_OCTETS) && sessionExpiryIds.containsKey(
                         case MqttSessionSignalFW.KIND_EXPIRY:
                             final MqttExpirySignalFW expirySignal = sessionSignal.expiry();
                             long expireAt = expirySignal.expireAt();
-                            final String16FW expiryClientId = expirySignal.clientId();
+                            final String16FW clientId = expirySignal.clientId();
+                            final String16FW expiryClientId =
+                                new String16FW().wrap(clientId.buffer(), clientId.offset(), clientId.limit());
```

Comment:
Creating a new flyweight does not make a copy of the underlying buffer.
```suggestion
                            final String16FW clientId = new String16FW(expirySignal.clientId().asString());
```